File: 24/day-2/solve.py
p = "input.txt"

with open(p, "rt") as f:
    arr = [[int(e) for e in x.split()] for x in f.read().splitlines()]

ans = 0
for l in arr:
    k = 0
    for e in range(len(l)):
        ld = l[:e] + l[e + 1 :]
        add = True
        b = ld[0] < ld[1]
        mn = 1 if b else -3
        mx = 3 if b else -1
        for i in range(len(ld) - 1):
            s = ld[i + 1] - ld[i]
            if s < mn or s > mx:
                add = False
        k = max(k, 1 if add else 0)
    ans += k

print(ans)

File: 24/day-1/solve.cpp
#include <bits/stdc++.h>
#include <mach-o/dyld.h>
using namespace std;

#define rep(i, a, b) for (int i = a; i < (b); ++i)
#define all(x) begin(x), end(x)
#define sz(x) (int)(x).size()
typedef long long ll;
typedef pair<int, int> pii;
typedef vector<int> vi;

string file = "input.txt";

string getParentPath()
{
    char buffer[PATH_MAX];
    uint32_t size = sizeof(buffer);
    if (_NSGetExecutablePath(buffer, &size) == 0)
    {
        char *lastSlash = strrchr(buffer, '/');
        if (lastSlash != nullptr)
            *lastSlash = '\0';
    }
    return buffer;
}

int main()
{
    ll ans = 0;
    int pre = -1, pre2 = -1;
    vi arr;
    unordered_multiset<int> ms;
    ifstream inputFile(getParentPath() + "/" + file);
    string line;
    while (getline(inputFile, line))
    {
        istringstream stream(line);
        int a, b;
        stream >> a >> b;
        pre = a;
        pre2 = b;
        arr.push_back(a);
        ms.insert(b);
    }
    rep(i, 0, sz(arr))
        ans += ms.count(arr[i]) * arr[i];
    cout << ans;
}
